src/FffGcodeWriter.cpp

```diff
@@ -2043,6 +2044,11 @@ void FffGcodeWriter::finalize()
     {
         gcode.writeMaxZFeedrate(getSettingInMillimetersPerSecond("machine_max_feedrate_z"));
     }
+
+    // set extrusion mode back to "normal"
+    const bool set_relative_extrusion_mode = (gcode.getFlavor() == EGCodeFlavor::REPRAP);
```

Comment:
Hmm come to think of it: I think this should be:
```
gcode.writeExtrusionMode(false);
gcode.finalize(...);
// set extrusion mode back to "normal"
const bool set_relative_extrusion_mode = (gcode.getFlavor() == EGCodeFlavor::REPRAP);
gcode.writeExtrusionMode(set_relative_extrusion_mode);
```

So that the start gcode and end gcode are always in absolute mode, but we return to 'normal' after the print.